src/phoenix/config.py

```diff
@@ -331,5 +353,85 @@ def get_web_base_url() -> str:
     return get_base_url()
 
 
+class LoggingMode(Enum):
+    AS_LIBRARY = "default"
+    AS_APPLICATION = "structured"
+
+
+def get_env_logging_mode() -> LoggingMode:
+    logging_mode = os.getenv(ENV_LOGGING_MODE) or LoggingMode.AS_LIBRARY.value
+    if (logging_mode_lower := logging_mode.lower()) in LoggingMode._value2member_map_:
+        return LoggingMode(logging_mode_lower)
+    raise ValueError(
+        f"Invalid value `{logging_mode}` for env var `{ENV_LOGGING_MODE}`. "
+        f"Valid values are: {log_a_list([mode.value for mode in LoggingMode],'and')} "
+        "(case-insensitive)."
+    )
+
+
+def get_env_logging_level() -> int:
+    return _get_logging_level(
+        env_var=ENV_LOGGING_LEVEL,
+        default_level=logging.INFO,
+    )
+
+
+def get_env_db_logging_level() -> int:
+    return _get_logging_level(
+        env_var=ENV_DB_LOGGING_LEVEL,
+        default_level=logging.WARNING,
+        # default_level=logging.INFO,
+    )
+
+
+def _get_logging_level(env_var: str, default_level: int) -> int:
+    logging_level = os.getenv(env_var)
+    # Default to INFO
+    if not logging_level:
+        return default_level
+
+    # levelNamesMapping = logging.getLevelNamesMapping() is not supported in python 3.8
+    # but is supported in 3.12. Hence, we get CI type errors if we don't used type:ignore for 3.8
+    # but we also get them fro 3.12 if we do use type:ignore.
+    # To go around this, we define the mapping ourselves and will remove this once we drop support
+    # for older python versions
+    levelNamesMapping = {
+        "CRITICAL": logging.CRITICAL,
+        "FATAL": logging.FATAL,
+        "ERROR": logging.ERROR,
+        "WARN": logging.WARNING,
+        "WARNING": logging.WARNING,
+        "INFO": logging.INFO,
+        "DEBUG": logging.DEBUG,
+        "NOTSET": logging.NOTSET,
+    }
+
+    valid_values = [level for level in levelNamesMapping if level != "NOTSET"]
+
+    if logging_level.upper() not in valid_values:
+        raise ValueError(
+            f"Invalid value `{logging_level}` for env var `{env_var}`. "
+            f"Valid values are: {log_a_list(valid_values,'and')} (case-insensitive)."
```

Comment:
```suggestion
            f"Valid values are: {"and, ".join(valid_values)} (case-insensitive)."
```

Not sure `log_a_list` is any simpler.

Reply:
For me the value in it is that it handles cases where the list is empty so you don't have to. In addition, it adds the "and" only at the end. For example: "a, b, and c" vs "a and b and c"

Reply:
I find it nice, but not attached to it